src/EnergyPlus/ExternalInterface.cc

```diff
@@ -871,13 +871,13 @@ void InstantiateInitializeFMUImport(EnergyPlusData &state)
     // Using/Aliasing
 
     // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-    int i, j; // Loop counters
 
     // Instantiate FMUs
-    for (i = 1; i <= state.dataExternalInterface->NumFMUObjects; ++i) {
-        for (j = 1; j <= state.dataExternalInterface->FMU(i).NumInstances; ++j) {
+    for (int i = 1; i <= state.dataExternalInterface->NumFMUObjects; ++i) {
+        for (int j = 1; j <= state.dataExternalInterface->FMU(i).NumInstances; ++j) {
+            auto folderStr = state.dataExternalInterface->FMU(i).Instance(j).WorkingFolder.string();
             state.dataExternalInterface->FMU(i).Instance(j).fmicomponent =
-                fmiEPlusInstantiateSlave((char *)state.dataExternalInterface->FMU(i).Instance(j).WorkingFolder.c_str(),
+                fmiEPlusInstantiateSlave((char *)folderStr.c_str(),
```

Comment:
Actual fix for #9313 . I used clang-query to find all calls to fs::path::c_str and made sure all others are fine.